lib/rules/no-import-compiler-macros.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @author Wayne Zhang
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const COMPILER_MACROS = new Set([
+  'defineProps',
+  'defineEmits',
+  'defineExpose',
+  'withDefaults'
+])
+
+const VUE_MODULES = new Set(['@vue/runtime-core', '@vue/runtime-dom', 'vue'])
+
+/**
+ * @param {Token} node
+ */
+function isComma(node) {
+  return node.type === 'Punctuator' && node.value === ','
+}
+
+/**
+ * @param {Token} node
+ */
+function isLeftCurlyBrace(node) {
+  return node.type === 'Punctuator' && node.value === '{'
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow importing vue compiler macros',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-import-compiler-macros.html'
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      noImportCompilerMacros:
+        "'{{name}}' is a compiler macro and no longer needs to be imported from '{{source}}'."
```

Comment:
```suggestion
        "'{{name}}' is a compiler macro and doesn't need to be imported."
```

I think "from {{ source }}" sounds like it should be imported from somewhere else. And "no longer" sounds like it had to be imported at some time, but I think it was never meant to be imported, was it?

Reply:
It never needs to be imported. The "no longer" stuff was directly copied from Vue's console warning..